requests/sessions.py

```diff
@@ -283,32 +295,21 @@ def rebuild_proxies(self, prepared_request, proxies):
         :rtype: dict
         """
         proxies = proxies if proxies is not None else {}
-        headers = prepared_request.headers
+        new_proxies = proxies.copy()
         url = prepared_request.url
         scheme = urlparse(url).scheme
-        new_proxies = proxies.copy()
-        no_proxy = proxies.get('no_proxy')
+        if scheme in proxies:
+            return new_proxies
 
-        bypass_proxy = should_bypass_proxies(url, no_proxy=no_proxy)
-        if self.trust_env and not bypass_proxy:
-            environ_proxies = get_environ_proxies(url, no_proxy=no_proxy)
+        no_proxy = proxies.get('no_proxy')
 
-            proxy = environ_proxies.get(scheme, environ_proxies.get('all'))
+        environ_proxies = get_environ_proxies(url, no_proxy=no_proxy)
```

Comment:
The changes here were done mainly to avoid calling `should_bypass_proxies` if not necessary because it is more of a bottleneck than `get_environ_proxies`

Reply:
IMHO this should be inside / under the `self.trust_env` code path.